src/coreclr/jit/morph.cpp

```diff
@@ -1062,7 +1062,6 @@ void CallArgs::ArgsComplete(Compiler* comp, GenTreeCall* call)
                         break;
                 }
```

Comment:
This logic essentially detects the cases that would require shifts and arithmetic to load the argument into registers without issuing too wide loads. Correct?

It would be helpful if this was the comment about all this logic. I'm not totally sure why the 11,13,14,15 cases do not have the same logic as the other case (i.e. allow it for locals) -- do you know why?

I think the entire thing would be simpler as
```cpp
unsigned lastLoadSize = structSize % REGSIZE_BYTES;
if (!isPow2(lastLoadSize) && !argObj->Addr()->IsLocalAddrExpr())
{
    SetNeedsTemp(&arg);
}
```
if I'm not missing something.

Reply:
> This logic essentially detects the cases that would require shifts and arithmetic to load the argument into registers without issuing too wide loads. Correct?

Yes. This also used to be necessary because codegen couldn't handle some mis-sized stack args.

(Though I like to think of it as mirroring the limitations of `fgMorphMultiregStructArg`)

> I'm not totally sure why the 11,13,14,15 cases do not have the same logic as the other case (i.e. allow it for locals) -- do you know why?

No, but it seems like coincidence. This code was added in 9a31fc473fd06749a34e503f972c516784ff26ad, then 0efa23b41c233d926d22d98aeeaa28b5cb63c9f8 added the code for small structs with the `IsLocalAddrExpr` condition.

> `!isPow2(lastLoadSize) && !argObj->Addr()->IsLocalAddrExpr()`

It's almost equivalent. There would be one pessimization: for mis-sized split structs larger than 16 bytes on ARM.

Will change it and add a comment.